src/native/corehost/hostmisc/utils.h

```diff
@@ -120,8 +120,9 @@ template<typename T>
 size_t to_size_t_dbgchecked(T value)
 {
     assert(value >= 0);
-    assert(value < static_cast<T>(std::numeric_limits<size_t>::max()));
-    return static_cast<size_t>(value);
+    size_t result = static_cast<size_t>(value);
+    assert(result >= 0);
+    return result;
```

Comment:
How does this work? `size_t` is unsigned, so technically it should always be `>=0`. I know that C++ is weird sometimes, but this code seems even weirder than usual :smile:.

Reply:
The assert was added while I was working on this PR, I assume as a response to some overflow bug. However the assert had a bug and started firing in valid cases. I tried to fix the assert. It does not look right again though. :-)

I think the idea is to check for overflow in the cast in a generic way. Not sure if that is even possible in a completely general case. Maybe as long as the cast does not lose the value and it can round-trip it, it is good enough for our purposes. I will try to change it to smth like that.

Reply:
The original `assert(value < static_cast<T>(std::numeric_limits<size_t>::max()));` obviously does not work with `int64` since max size_t when cast to T would be a negative value.